File: python/locations.py
# locations.py
import os
from API import BraveAPI
from utils.main import dump_schema

# Create an instance of the BraveAPI class
brave = BraveAPI({
    'subscription_token': os.environ['BRAVE_KEY']
})

# Setting up parameters
params = {
    'q': "jiu jitsu gyms in san francisco",
    'freshness': "pw",
}

# Performing the search
results = brave.search(params)

# Dumping and printing the schema of the results
print(dump_schema(results['locations']))

File: js/locations.js
// locations.js
const { dumpSchema } = require("./utils/index.js");
const BraveAPI = require("./API.js");

// Create a new instance of the BraveAPI class
const brave = new BraveAPI({
    subscription_token: process.env.BRAVE_KEY
});

// Setup the search parameters
const params = {
    q: "jiu jitsu gyms in san francisco",
    freshness: "py",
};

// Perform the search
brave.search(params).then(results => {
    // Log the results
    console.log(dumpSchema(results.locations));
});

